tests/components/dsmr/test_sensor.py

```diff
@@ -47,6 +49,34 @@ async def connection_factory(*args, **kwargs):
     return connection_factory, transport, protocol
 
 
+async def test_setup_platform(hass, mock_connection_factory):
+    """Test setup of platform."""
+    async_add_entities = MagicMock()
+
+    entry_data = {
+        "platform": DOMAIN,
+        "port": "/dev/ttyUSB0",
+        "dsmr_version": "2.2",
+        "precision": 4,
+        "reconnect_interval": 30,
+    }
+
+    assert await async_setup_component(hass, SENSOR_DOMAIN, {SENSOR_DOMAIN: entry_data})
```

Comment:
i think we could patch async_entry and async_setup_entry here since we really dont need it to create all the entities. just go confirm creation of config entry.